nautobot/extras/api/views.py

```diff
@@ -283,11 +284,16 @@ def run(self, request, class_path):
         input_serializer = serializers.JobInputSerializer(data=request.data)
         input_serializer.is_valid(raise_exception=True)
 
-        data = input_serializer.data["data"]
+        data = input_serializer.data["data"] or {}
         commit = input_serializer.data["commit"]
         if commit is None:
             commit = getattr(job_class.Meta, "commit_default", True)
 
+        try:
+            job.validate_data(data)
+        except FormsValidationError as e:
+            return Response({"errors": e.message_dict if hasattr(e, "error_dict") else e.messages}, status=400)
```

Comment:
Just checking here - is the asymmetry between `e.message_dict` and `hasattr(e, "error_dict")` correct and intentional? Or should they both be either `message_dict` or `error_dict`?

Reply:
Sadly this is intentional. `ValidationError.message_dict` is a property that can only be accessed when `ValidationError.error_dict` was preset (i.e. `ValidationError` was passed a dict in the constructor). Otherwise we can only get to the list of errors using `ValidationError.messages`. This is what I meant when I said that `ValidationError` is slightly uncomfortable to work with.

Reply:
Ok, thanks! Maybe add a comment documenting this for future reference?

Reply:
Done!